scvi/module/_mixupvae.py

```diff
@@ -334,14 +334,27 @@ def inference(
             encoder_pseudobulk_input = x_pseudobulk_
             encoder_signature_input = x_signature_
 
-        # categorical covariates
+        ## Intialize empty tuples for categorical covariates
+        # Categorical covariates for output dict
+        categorical_input = ()
+        categorical_pseudobulk_input = ()
+        categorical_signature_input = ()
+        # Categorical covariates for encoder
+        categorical_input_encoder = ()
+        categorical_pseudobulk_input_encoder = ()
+        categorical_signature_input_encoder = ()
+
+        # /!\ Categorical covariates should *always* be returned for the decoder
         if cat_covs is not None:
             cat_covs = torch.split(cat_covs, 1, dim=1)
             categorical_input = []
             categorical_pseudobulk_input = []
             categorical_signature_input = []
             j=0
-            for n_cat in self.z_encoder.encoder.n_cat_list:
+            # Use the decoder's `n_cat_list` arg to determine the number
+            # of categories. The encoder's `n_cat_list` will be set to `None` if
+            # `self.encoder.encode_covariates`` is set to `False``
+            for n_cat in self.decoder.px_decoder.n_cat_list:
                 if n_cat > 0 :
```

Comment:
@SimonGrouard Important detail I just noticed:

If `self.encoder_covariates=False` we are not passing the categorical covariates to the decoder even when `cat_covs != None` because of this [line](https://github.com/owkin/deepdeconv-fork/blob/ba51f7c5db60e57b8328fffb23e8fc32ea59bd54/scvi/module/_vae.py#L178C9-L178C67) in the initialization of the encoder:

I updated the `inference` so that we always output the categorical covariates that will be passed to the decoder.